### PR TITLE
Compact xrefs

### LINKED ISSUE #10253: Compact xrefs
```
[0x10000443a]> pdf
╭ (fcn) sym.imp.__error 6
│   sym.imp.__error ();
│           ; CALL XREF from 0x100001da0 (sym.func.100001b44 + 604)
│           ; CALL XREF from 0x100001db0 (sym.func.100001b44 + 620)
│           ; CALL XREF from 0x100001db8 (sym.func.100001b44 + 628)
│           ; CALL XREF from 0x100002eaa (sym.func.100002aad + 1021)
╰           0x10000443a      ff25080c0000   jmp qword [0x100005048]    ; [0x100005048:8]=0x10000464e
[0x10000443a]>
```

in this example we can find that there are 2 function xrefs, one of them with 3 xrefs.

we can compact the xrefs in verbose mode if we use a syntax like this:

```
; CALL XREF from sym.func.0x100001b44 (..1da0, ..1db0, ..2eaa)
; CALL XREF from sym.func.100002aad (0x100002eaa)
```

maybe its more readable?

pls discuss

### PR DESCRIPTION
**Pull request by cyanpencil**
Closes #10253 

2018-07-12-040420_682x176_scrot

### CODE REVIEW

File: libr/core/disasm.c [hunk 1]
```diff
@@ -1182,15 +1183,36 @@ static void ds_show_xrefs(RDisasmState *ds) {
...
-			RAnalFunction *fun = fcnIn (ds, refi->addr, -1); //r_anal_get_fcn_in (core->anal, refi->addr, -1);
+			fun = fcnIn (ds, refi->addr, -1); 
 			if (fun) {
 				name = strdup (fun->name);
+				if (iter != xrefs->tail) {
+					ut64 next_addr = ((RAnalRef *)(iter->n->data))->addr;
+					next_fun = r_anal_get_fcn_in (core->anal, next_addr, -1);
+					if (next_fun && next_fun->addr == fun->addr) {
+						r_list_append (addrs, refi->addr);
+						continue;
```
Comment (ret2libc) [inline libr/core/disasm.c]: `name` is not freed

File: libr/core/disasm.c [hunk 2]
```diff
@@ -1182,15 +1183,36 @@ static void ds_show_xrefs(RDisasmState *ds) {
...
+				f = r_flag_get_at (core->flags, refi->addr, true);
 				if (f) {
-					name = r_str_newf ("%s + %d", f->name, refi->addr - f->offset);
+					name = strdup (f->name);
+					if (iter != xrefs->tail) {
+						ut64 next_addr = ((RAnalRef *)(iter->n->data))->addr;
+						next_f = r_flag_get_at (core->flags, next_addr, true);
+						if (next_f && f->offset == next_f->offset) {
+							r_list_append (addrs, refi->addr - f->offset);
+							continue;
```
Comment (ret2libc) [inline libr/core/disasm.c]: `name` is not freed

**Comment by cyanpencil** [comment]
Thanks for the feedback @ret2libc ,
you are absolutely right about the missing free, I'll fix it right now.
I'll try also to refactor the two similar pieces into one, but I'm not very sure on how to do that right now ^^

**Comment by ret2libc** [comment]
Yeah don't worry about the refactoring.. It was just a comment, but I guess for the moment it's ok as it is... I don't see an easy way to refator this either

**Pull request merged**